eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/ConfigResolver.php

```diff
@@ -237,8 +245,95 @@ public function getDefaultScope()
         return $this->defaultScope ?: $this->siteAccess->name;
     }
 
+    /**
+     * @param string $scope The default "scope" aka siteaccess name, as opposed to the self::SCOPE_DEFAULT.
+     */
     public function setDefaultScope($scope)
     {
         $this->defaultScope = $scope;
+
+        // On scope change check if siteaccess has been updated so we can log warnings if there are any
+        if ($this->siteAccess->matchingType !== 'uninitialized') {
+            $this->warnAboutTooEarlyLoadedParams();
+        }
+    }
+
+    private function warnAboutTooEarlyLoadedParams()
+    {
+        if (empty($this->tooEarlyLoadedList)) {
+            return;
+        }
+
+        $logger = $this->container->get('logger');
+        foreach ($this->tooEarlyLoadedList as $param => $services) {
+            $logger->warning(sprintf(
+                'ConfigResolver was used to load parameter "%s" before SiteAccess was loaded by services: %s. This can cause issues. '
+                . 'Try to use ConfigResolver lazily, '
+                . (PHP_SAPI === 'cli' ? 'make commands that rely on them lazy, ' : '')
```

Comment:
❗️ MERGE NOTE ❗️ : This is only for 7.5, as only Symfony 3.4 supports making commands lazy _(aka allows setting command name on service tag so it does not need to load command to get that info)_.